### ISSUE TITLE
Invalid support in implementation of the default methods.

### ISSUE THREAD

**Issue by gallandarakhneorg** [eclipse/xtext-xtend#191]
Let the following Xtend code:
```
class TestMultiInheritance<T> extends AbstractCollection<T> implements Set<T> {
	
	override iterator() {
		spliterator
                null
	}
	
	override size() {
               0
	}
	
}
```

The Xtend compiler generates the error "The type TestMultiInheritance inherits multiple implementations of the method spliterator() from Iterable and Set."

However, the equivalent Java code is compiling:
```java
public class TestMultiInheritance<T extends Object> extends AbstractCollection<T> implements Set<T> {
  public Iterator<T> iterator() {
    this.spliterator();
    return null;
  }
  
  public int size() {
    return 0;
  }
}
```

The spliteraor function is defined in the Collection and Set interfaces with different code. The Java compiler takes the default implementation from Set because is it the closest definition from the call in the type hierarchy.

Is a feature of Xtend to fail in this case? Or should Xtend be fixed in order to have a similar behavior as the Java compiler?

**Comment by cdietrich** [comment]
looks like a bug in the return type inference to me.

```
class TestMultiInheritance<T> extends AbstractCollection<T> implements Set<T> {
	
	override Iterator<T> iterator() {
		spliterator
                null
	}
	
	override size() {
               0
	}
	
}
```
seems to work

**Comment by cdietrich** [comment]
wthat makes me wonder: which super method wins in java?

**Comment by gallandarakhneorg** [comment]
The one coming from the just implemented interface. As far as I remember, the heuristic is based on the distance between the call and the default method definition into the type hierarchy. The definition of spliterator in Set is closer to the definition of spliterator in Collection or Iterable (provided by the extended type).

I just created the example in my Eclipse for DSL developer (not my own DSL product), and the issue is here.

**Comment by cdietrich** [comment]
yes unfortunately we have tons of bugs around handling of default methods

**Comment by gallandarakhneorg** [comment]
I will search for a fix and create a PR if I found something suitable.
